File: src/main/java/io/rsocket/cli/http2/Http2UriHandler.java
package io.rsocket.cli.http2;

import static java.util.Optional.of;

import io.rsocket.transport.ClientTransport;
import io.rsocket.uri.UriHandler;
import org.eclipse.jetty.http.HttpScheme;

import java.net.URI;
import java.util.Optional;

public class Http2UriHandler implements UriHandler {
  @Override
  public Optional<ClientTransport> buildClient(URI uri) {
    if (HttpScheme.HTTPS.is(uri.getScheme()) || HttpScheme.HTTP.is(uri.getScheme())) {
      return of(new Http2ClientTransport(uri));
    }

    return UriHandler.super.buildClient(uri);
  }
}

File: src/main/kotlin/io/rsocket/cli/http2/Http2UriHandler.kt
package io.rsocket.cli.http2

import io.rsocket.cli.uri.UriHandler
import io.rsocket.transport.ClientTransport
import io.rsocket.transport.ServerTransport
import org.eclipse.jetty.http.HttpScheme
import java.net.URI
import java.util.Optional
import java.util.Optional.empty
import java.util.Optional.of

class Http2UriHandler : UriHandler {
  override fun buildClient(
    uri: URI,
    headerMap: Map<String, String>
  ): Optional<ClientTransport> =
    when {
      HttpScheme.HTTPS.`is`(uri.scheme) || HttpScheme.HTTP.`is`(uri.scheme) -> of(Http2ClientTransport(uri))
      else -> empty()
    }

  override fun buildServer(uri: URI): Optional<ServerTransport<*>> = empty()
}
